Update TypeScript Express to use CLI for creating Okta app (#642)
oktadeveloper/okta-blog: _source/_posts/2018-11-15-node-express-typescript.md

- "Build a @nodejs application using @typescriptlang and #expressjs"
image: blog/node-express-typescript/node-express-typescript.jpg
type: conversion
changelog: 

>   - 2021-03-30: Updated to use CLI for creating Okta application.

Reviewer: I've been adding changelogs when I updated libraries. Otherwise, I've left it out. If you leave it in, I'd recommend linking to this PR so people can see what changed.

```
You can see what's changed in [okta-blog#642](https://github.com/oktadeveloper/okta-blog/pull/642).
```

## Easily Add Authentication to Node and Express

Adding user registration and login (authentication) to any application is not a trivial task. The good news is Okta makes this step very easy. 

> {% include setup/cli.md type="web" loginRedirectUri="http://localhost:8080/authorization-code/callback" %}

Reviewer: This isn't rendering the redirect URL correctly. I'm looking into how to fix now.

Adding user registration and login (authentication) to any application is not a trivial task. The good news is Okta makes this step very easy. 

{% img blog/node-express-typescript/add-application-00.jpg alt:"Sign up for free account" width:"800" %}{: .center-image }

After creating your account, click the **Applications** link at the top, and then click **Add Application**.

{% img blog/node-express-typescript/add-application-01.jpg alt:"Create application" width:"500" %}{: .center-image }

Next, choose a **Web Application** and click **Next**.

{% img blog/node-express-typescript/add-application-02.jpg alt:"Create a web application" width:"800" %}{: .center-image }

Enter a name for your application, such as *Guitar Inventory*. Verify the port number is the same as configured for your local web application. Then, click **Done** to finish creating the application.

> {% img blog/node-express-typescript/add-application-03.jpg alt:"Application settings" width:"800" %}{: .center-image }

Reviewer: If you look at this image, you'll see it uses one of the old default redirect URIs (that's now gone). The CLI tends to have different defaults than the admin console, so we need to specify what's in this image. You can do that with the `loginRedirectUri` attribute.

```
{% include setup/cli.md type="web" loginRedirectUri="http://localhost:8080/authorization-code/callback" %}
```

> {% include setup/cli.md type="web" framework="Node" %}

Reviewer: Unfortunately, Node is not yet an option. These are what it gives you after selecting "Web":

```
Enter your choice [Web]: 1
Type of Application
> 1: Okta Spring Boot Starter
> 2: Spring Boot
> 3: JHipster
> 4: Quarkus
> 5: Other
```

If you remove the `framework` attribute, it defaults to **Other**.

OKTA_CLIENT_SECRET={yourClientSecret}
```

### Enable self-service registration

One of the great features of Okta is allowing users of your application to sign up for an account. By default, this feature is disabled, but you can easily enable it. Run `okta login` to get the URL for your Okta org. Open the result in your favorite browser and log in to the Okta Admin Console.

> 1. Click on the **Users** menu and select **Registration**

Reviewer: ```suggestion
1. Click on the **Directory** menu and select **Self-Service Registration**
```

OKTA_CLIENT_SECRET={yourClientSecret}
```

### Enable self-service registration

> One of the great features of Okta is allowing users of your application to sign up for an account. By default, this feature is disabled, but you can easily enable it. 

Reviewer: ```suggestion
One of the great features of Okta is allowing users of your application to sign up for an account. By default, this feature is disabled, but you can easily enable it. Run `okta login` to get the URL for your Okta org. Open the result in your favorite browser and log in to the Okta Admin Console.
```

> 1. Click on the **Users** menu and select **Registration**

Reviewer: ```suggestion
1. Go to **Directory** and select **Self-Service Registration**
```

> 2. Click on the **Edit** button.

Reviewer: ```suggestion
2. Click on the **Enable Registration** button.
```

> 3. Change **Self-service registration** to *Enabled*.

Reviewer: ```suggestion
3. If you don't see this button, click **Edit** and change **Self-service registration** to *Enabled*.
```
